packages/fxa-settings/src/components/PageWithLoggedInStatusState/index.tsx

```diff
@@ -1,24 +1,41 @@
 import React, { useState, useEffect } from 'react';
-import { useInitialState } from '../../models';
+import { useInitialState, useRelier } from '../../models';
 import { RouteComponentProps } from '@reach/router';
 
 export const PageWithLoggedInStatusState = (
-  props: any & RouteComponentProps & { Page: React.ElementType }
+  props: any &
+    RouteComponentProps & {
+      Page: React.ElementType;
+    }
 ) => {
   const { Page } = props;
   const { loading, error } = useInitialState();
 
   const [isSignedIn, setIsSignedIn] = useState<boolean>();
+  const [isSync, setIsSync] = useState<boolean>();
+  const [serviceName, setServiceName] = useState<string>();
+  const relier = useRelier();
+
+  // TODO: Get the broker `continue` action once https://mozilla-hub.atlassian.net/browse/FXA-6989 is merged
+  let continueHandler: Function | undefined;
 
   useEffect(() => {
     if (!loading && error?.message.includes('Invalid token')) {
       setIsSignedIn(false);
     } else if (!loading && !error) {
       setIsSignedIn(true);
     }
-  }, [error, loading]);
+    try {
+      relier.isSync().then((res: boolean) => {
```

Comment:
Can we use `await` here similarly to how we've set other `async` functions in a `useEffect`?

I'd have to dig, but can `isSync` ever be truthy if `serviceName` doesn't contain `Sync`? I guess I'm wondering if we can just pass along `serviceName`, and then when we need to check `isSync` we check against the service name (which we do in content-server in a couple of different spots) but maybe it's more complex than that?